experimental/packages/opentelemetry-instrumentation-http/src/http.ts

```diff
@@ -307,7 +307,7 @@ export class HttpInstrumentation extends InstrumentationBase<Http> {
           this._diag.debug('outgoingRequest on end()');
           let status: SpanStatus;
 
-          if (response.aborted && !response.complete) {
+          if (response.aborted && !response.complete && span.status['code'] !== SpanStatusCode.ERROR) {
```

Comment:
An example of using properties not available in the API span type.

Reply:
Do you think making those properties available in the API span type makes sense? I imagine that importing both of them is not a good practice.

Reply:
They are intentionally left out of the API. If you think they should be exposed via the API you would need to make that argument in the specification, not here. We get a lot of leeway in terms of implementation details, but typically if something isn't specified we won't add it to the API.

Reply:
I'm not sure I actually see what you're trying to fix. Can you provide a minimal example of the problem?

Reply:
If a span has the status set as Error and its message, any call to `setStatus` to set it to error again will overwrite anything that was there before. This PR prevents that from happening.

Reply:
How does a span created by the http instrumentation get an error status except by the http instrumentation? Surely this is more likely to be a wrongly set error status?

Reply:
You can get that span requesting the active one with the tracer provider and try to add things related to your application. One of these things can be an exception and some status changes.

Reply:
I see. So as a user you are trying to override the behavior specified here: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#status ?

Reply:
Yes, in this case, I'm specifically talking about an Express error handler. When a 5XX happens, I want to reflect that in the active span changing its status, which happens to be the HTTP span. If I do so, when the span is closed by the HTTP instrumentation, it ignores the previously set Error status by my error handler and deletes the message.

Reply:
Hmm ok I see. To my understanding, a 5XX response should be set to Error by the http instrumentation anyway, but the general problem of two instrumentations overriding the same status is quite a tricky one. The conceptual model of opentelemetry doesn't really account for the idea that multiple instrumentations would modify the same span. That is just something we did with the express instrumentation to overcome some other limitations.

As far as the fix, we cannot use SDK-only methods on the span here. There could be other types of span like NoopSpan and NonRecordingSpan which don't have those properties and would fail. I see two options: open a spec issue to add a status getter to the API, or do some type of instanceof check to see if we have an SDK span before using SDK properties.

Reply:
I'll try with the type checking and will also make some changes to the logic to fit better with the current spec:

https://opentelemetry.io/docs/reference/specification/trace/api/#set-status